src/Build/Instance/ProjectItemInstance.cs

```diff
@@ -28,7 +28,13 @@ namespace Microsoft.Build.Execution
     /// and evaluation has already been performed, so it is unnecessary bulk.
     /// </remarks>
     [DebuggerDisplay("{ItemType}={EvaluatedInclude} #DirectMetadata={DirectMetadataCount})")]
-    public class ProjectItemInstance : IItem<ProjectMetadataInstance>, ITaskItem2, IMetadataTable, ITranslatable, IDeepCloneable<ProjectItemInstance>
+    public class ProjectItemInstance :
```

Comment:
For consistency's sake, worth making `ProjectItem` implement `IMetadataContainer` well? I bet CPS will appreciate the zero allocation member.

Reply:
This opens a whole 'nother can of worms :) I started on this but quickly ran into viral things such as ProjectItemLink etc. I think this should be a separate PR ideally driven by an actual scenario that we can test and tune.